app/validators/value_source_validator.py

```diff
@@ -26,6 +55,118 @@ def __init__(self, value_source, json_path, questionnaire_schema):
             "calculated_summary": self.questionnaire_schema.calculated_summary_block_ids,
         }
 
+        if self.value_source["source"] == "progress":
+            self._valid_source_identifiers_map["progress"] = {
+                "block_ids": self._get_valid_progress_value_source_block_identifiers(),
+                "section_ids": self._get_valid_progress_value_source_section_identifiers(),
+            }
+
+    def _get_valid_progress_value_source_block_identifiers(self):
+        valid_block_ids = (
+            self.past_block_ids
+            - set([self.current_block_id])
+            - self.block_ids_in_past_repeating_sections
+        )
+
+        return valid_block_ids
+
+    def _get_valid_progress_value_source_section_identifiers(self):
+        valid_section_ids = (
```

Comment:
Can be updated similarly to the suggestion above

Reply:
updated